packages/map/src/index.ts

```diff
@@ -48,73 +43,90 @@ export class ReactiveMap<K, V> extends Map<K, V> {
       this.#keyTriggers.track(key);
       yield key;
     }
+
     this.#keyTriggers.track($KEYS);
   }
+
   *values(): IterableIterator<V> {
-    for (const [key, v] of super.entries()) {
+    for (const [key, value] of super.entries()) {
       this.#valueTriggers.track(key);
-      yield v;
+      yield value;
     }
+
     this.#keyTriggers.track($KEYS);
   }
+
   *entries(): IterableIterator<[K, V]> {
     for (const entry of super.entries()) {
+      this.#keyTriggers.track(entry[0]);
       this.#valueTriggers.track(entry[0]);
       yield entry;
     }
+
+    this.#keyTriggers.track($KEYS);
+  }
+
+  forEach(fn: (value: V, key: K, map: Map<K, V>) => void): void {
     this.#keyTriggers.track($KEYS);
+
+    for (const [key, value] of super.entries()) {
+      this.#keyTriggers.track(key);
+      this.#valueTriggers.track(key);
+      fn(value, key, this);
+    }
+  }
+
+  has(key: K): boolean {
+    this.#keyTriggers.track(key);
+    return super.has(key);
+  }
+
+  get(key: K): V | undefined {
+    this.#valueTriggers.track(key);
+    return super.get(key);
   }
 
-  // writes
   set(key: K, value: V): this {
+    const hasKey = super.has(key);
+    const currentValue = super.get(key);
+    const result = super.set(key, value);
```

Comment:
Why bring them here, whats wrong with them being inlined? Is this important  or just a preference?